vendor/bat-native-confirmations/src/bat/confirmations/internal/ads_rewards.cc

```diff
@@ -0,0 +1,265 @@
+    payments_(std::make_unique<Payments>(confirmations, confirmations_client)),
+    ad_grants_(std::make_unique<AdGrants>(confirmations, confirmations_client)),
+    confirmations_(confirmations),
+    confirmations_client_(confirmations_client) {
+  BLOG(INFO) << "Initializing ads rewards";
+}
+
+AdsRewards::~AdsRewards() {
+  BLOG(INFO) << "Deinitializing ads rewards";
+
+  CancelRetry();
+}
+
+void AdsRewards::Fetch(const WalletInfo& wallet_info) {
+  DCHECK(!wallet_info.payment_id.empty());
+  DCHECK(!wallet_info.public_key.empty());
+
+  BLOG(INFO) << "Fetch ads rewards";
+
+  wallet_info_ = WalletInfo(wallet_info);
+
+  GetPaymentBalance();
+}
+
+bool AdsRewards::OnTimer(const uint32_t timer_id) {
+  BLOG(INFO) << "OnTimer: " << std::endl
+      << "  timer_id: " << std::to_string(timer_id) << std::endl
+      << "  retry_timer_id_: " << std::to_string(retry_timer_id_) << std::endl;
+
+  if (timer_id == retry_timer_id_) {
+    Retry();
+
+    return true;
+  }
+
+  return false;
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+void AdsRewards::GetPaymentBalance() {
+  BLOG(INFO) << "GetPaymentBalance";
+
+  BLOG(INFO) << "GET /v1/confirmation/payment/{payment_id}";
+  GetPaymentBalanceRequest request;
+
+  BLOG(INFO) << "URL Request:";
+
+  auto url = request.BuildUrl(wallet_info_);
+  BLOG(INFO) << "  URL: " << url;
+
+  auto method = request.GetMethod();
+
+  auto body = request.BuildBody();
+  BLOG(INFO) << "  Body: " << body;
+
+  auto headers = request.BuildHeaders(body, wallet_info_);
+  BLOG(INFO) << "  Headers:";
+  for (const auto& header : headers) {
+    BLOG(INFO) << "    " << header;
+  }
+
+  auto content_type = request.GetContentType();
+  BLOG(INFO) << "  Content_type: " << content_type;
+
+  auto callback = std::bind(&AdsRewards::OnGetPaymentBalance,
+      this, url, _1, _2, _3);
+
+  confirmations_client_->LoadURL(url, headers, "", "", method, callback);
+}
+
+void AdsRewards::OnGetPaymentBalance(
+    const std::string& url,
+    const int response_status_code,
+    const std::string& response,
+    const std::map<std::string, std::string>& headers) {
+  BLOG(INFO) << "OnGetPaymentBalance";
+
+  BLOG(INFO) << "URL Request Response:";
+  BLOG(INFO) << "  URL: " << url;
+  BLOG(INFO) << "  Response Status Code: " << response_status_code;
+  BLOG(INFO) << "  Response: " << response;
+  BLOG(INFO) << "  Headers:";
+  for (const auto& header : headers) {
+    BLOG(INFO) << "    " << header.first << ": " << header.second;
+  }
+
+  if (response_status_code != net::HTTP_OK) {
+    BLOG(ERROR) << "Failed to get payment balance";
+    OnAdsRewards(FAILED);
+    return;
+  }
+
+  if (!payments_->ParseJson(response)) {
+    BLOG(ERROR) << "Failed to parse payment balance: " << response;
+    OnAdsRewards(FAILED);
+    return;
+  }
+
+  GetAdGrants();
+}
+
+void AdsRewards::GetAdGrants() {
+  BLOG(INFO) << "GetAdGrants";
+
+  BLOG(INFO) << "GET /v2/wallet/{payment_id}/grants/ads";
```

Comment:
Should this also have the signature headers that `GetPaymentBalance()` adds via `request.BuildHeaders()`?

Reply:
This endpoint did not support the signature however we are making this change so it will before release